### PR TITLE
Clean up package version definitions using Directory.Packages.props

### LINKED ISSUE #4260: [repo] Clean up package version definitions using Directory.Packages.props
Today we have our package versions defined in props files mixed with other stuff:

* https://github.com/open-telemetry/opentelemetry-dotnet/blob/735dd0d04c755af7cc1ddfe7565c53757f36d5ab/build/Common.props#L24-L54
* https://github.com/open-telemetry/opentelemetry-dotnet/blob/735dd0d04c755af7cc1ddfe7565c53757f36d5ab/build/Common.nonprod.props#L28-L51

There is a new [NuGet feature](https://devblogs.microsoft.com/nuget/announcing-nuget-6-5-the-sweetest-way-to-manage-your-packages/#streamlined-package-management-inside-visual-studio-with-directory-packages-props-support) which allows this to be moved into `Directory.Packages.props` and adds tooling support.

Proposing we move all of these package version definitions into `Directory.Packages.props` and clean up the project files.

### PR DESCRIPTION
**Pull request by luizhlelis**
Fixes #4260
Design discussion issue #

## Changes

Clean up package version definitions using `Directory.Packages.props`.

- I'm creating a root `Directory.Packages.props` and others in specific folders to override some of the root package versions based on what I found in `Common.props` and `Common.nonprod.props`:

```
OpenTelemetry.sln
Directory.Packages.props
- docs
      Directory.Packages.props
- test
      Directory.Packages.props
- examples
      Directory.Packages.props
```

- Some properties are used in power shell scripts, like [MicrosoftCodeCoveragePkgVer](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/build/process-codecoverage.ps1#L2) and `OTelPreviousStableVer`, so I'm keeping them in `Common.props`;
- Other packages like `SerilogPkgVer` and `SystemCollectionsImmutablePkgVer` are not being used in any `.cspjoj` so I deleted their reference from `Common.props`.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: Directory.Packages.props
Comment (TimothyMothra) [inline Directory.Packages.props:7]: nit: consider using a separate ItemGroup for prod and non-prod.
Then should alphabetize the packages within those groups.

File: docs/trace/getting-started-jaeger/getting-started-jaeger.csproj
Comment (TimothyMothra) [inline docs/trace/getting-started-jaeger/getting-started-jaeger.csproj:8]: Red flag for me that the Condition was dropped.

File: Directory.Packages.props
Comment (luizhlelis) [reply Directory.Packages.props]: [fixed](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4298/commits/ecbbae97770b074f61c3b653223c41cb4d82b057)

File: docs/trace/getting-started-jaeger/getting-started-jaeger.csproj
Comment (luizhlelis) [reply docs/trace/getting-started-jaeger/getting-started-jaeger.csproj:8]: [fixed](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4298/commits/52c6481d73fce2ef28ff75c9091c47a787bc05fa)

**Comment by TimothyMothra** [comment]
LGTM!
Need one of the repo maintainers to queue the builds.
Assuming the builds pass, I would approve this.

File: NuGet.config
Comment (luizhlelis) [inline NuGet.config:20]: The following error was thrown after central package management because `opentelemetry.net` is using more than one package source:

```sh
error NU1507: Warning As Error: There are 2 package sources defined in your configuration.
When using central package management, please map your package sources with package
source mapping (https://aka.ms/nuget-package-source-mapping) or specify a single package source.
The following sources are defined: NuGet, .Net Core Tools 
```

There are two solutions:

- To ignore the warning (as error) with `<NoWarn>$(NoWarn);NU1507</NoWarn>`. I don't think this is the best option;
- Enable [package source mapping](https://learn.microsoft.com/en-us/nuget/consume-packages/package-source-mapping#enabling-package-source-mapping).

As I'm choosing to follow the second approach, I need to know what packages (or package name pattern) should be restored by the `.Net Core Tools` source.

Comment (luizhlelis) [reply NuGet.config:20]: FYI: @TimothyMothra

Comment (TimothyMothra) [reply NuGet.config:20]: I didn't know this repo had an additional PackageSource.
May need to open a new Issue to review this, this may not be needed anymore.

Looks like this was introduced in 2021 to use a 6.0.0-beta version of the ApiCompat tool. (https://github.com/open-telemetry/opentelemetry-dotnet/pull/2075/files)
However, this tool appears to be published publicly now and has newer versions. https://www.nuget.org/packages/Microsoft.DotNet.ApiCompat.Tool/

File: Directory.Packages.props
Comment (luizhlelis) [inline Directory.Packages.props:1]: Hello everyone.

Is there anything that I can do to unblock the PR?

File: test/OpenTelemetry.Instrumentation.Http.Tests/OpenTelemetry.Instrumentation.Http.Tests.csproj
Comment (alanwest) [inline test/OpenTelemetry.Instrumentation.Http.Tests/OpenTelemetry.Instrumentation.Http.Tests.csproj:21]: Condition was removed.

```suggestion
    <PackageReference Include="System.Net.Http" Condition="'$(TargetFramework)' == 'net462'" />
```

**Review by alanwest — COMMENTED** [review]
Hey @luizhlelis sorry for the delay in review. I see one more spot where the condition of a reference was dropped. Other than that, I think this PR looks good.

File: test/OpenTelemetry.Instrumentation.Http.Tests/OpenTelemetry.Instrumentation.Http.Tests.csproj
Comment (luizhlelis) [reply test/OpenTelemetry.Instrumentation.Http.Tests/OpenTelemetry.Instrumentation.Http.Tests.csproj]: thank you @alanwest! 🙂

**Review by alanwest — APPROVED** [review]
Thanks, @luizhlelis, this is a nice improvement!

**Pull request merged**
